Doc update for "Deploy RisingWave on Kubernetes with Helm" (#2001)
risingwavelabs/risingwave-docs: docs/deploy/deploy-k8s-helm.md

Now start a RisingWave cluster with Helm.

> 1. Add the Helm repo repository:

Reviewer: ```suggestion
1. Add the RisingWave Helm chart repository:
```

  ```bash
  helm repo add risingwavelabs https://risingwavelabs.github.io/helm-charts/ --force-update
  ```

2. Update your Helm chart repositories to ensure that you are using the RisingWave Helm chart:

  ```bash
  helm repo update
  ```

  If you are using AWS EKS, you also need to update the local configuration for kubectl and Helm to access your EKS cluster:

  ```bash
  aws eks update-kubeconfig --name <your_eks_cluster_name>
  ```

> 3. Create RisingWave namespace. We recommend creating a dedicated namespace for RisingWave resources.

Reviewer: ```suggestion
3. Create a RisingWave namespace. We recommend creating a dedicated namespace for RisingWave resources.
```

    - **Customize state store**: The state store in RisingWave serves as a fault-tolerant storage system for preserving system state. See [Configuration](https://github.com/risingwavelabs/helm-charts/blob/main/docs/CONFIGURATION.md#customize-state-store) for all the available options and [Examples](https://github.com/risingwavelabs/helm-charts/tree/main/examples/state-stores) for detailed usage of state stores.

    - **Bundled etcd and MinIO**: If you want to use `etcd` as the meta store and `MinIO` as the state store, the Helm chart for RisingWave offers the option to bundle them together. This allows for a quick and easy setup of the Helm chart. See [Configuration](https://github.com/risingwavelabs/helm-charts/blob/main/docs/CONFIGURATION.md#bundled-etcdminio-as-stores) for more details. To enable this feature, set `tags.bundle=true`.

  :::note
  Before using the bundled `etcd` and `MinIO`, and any local stores, ensure that you have implemented the [Dynamic Volume Provisioning](https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/).
  :::

> 5. During installation or upgrade, you also need to customize your RisingWave deployment by editing the [`values.yml`](https://github.com/risingwavelabs/helm-charts/blob/main/charts/risingwave/values.yaml) file. You should edit the file before specifying it during installation or upgrade.

Reviewer: @WanYixian Since step 4 is about editing the file (customizing the deployment), this step is about applying the deployment configuration during installation or upgrade.

> 5. During installation or upgrade, you also need to customize your RisingWave deployment by editing the [`values.yml`](https://github.com/risingwavelabs/helm-charts/blob/main/charts/risingwave/values.yaml) file. You should edit the file before specifying it during installation or upgrade.

Reviewer: I think this step is unnecessary. Step 4 is about customizing the deployment, and Step 6 is about specifying this configuration file during installation. This step is a note about when you should customize the file. This point can be mentioned in step 4.
We have another topic about upgrading RisingWave. Information about upgrading RW via K8s should be added there (I think we already have a big chunk of it, so please double check).

Author: Checked with Shunjie, delete step 5 now

Now start a RisingWave cluster with Helm.

### 1. Add RisingWave repo

Add the `risingwave` chart repository:

  ```bash
  helm repo add risingwavelabs https://risingwavelabs.github.io/helm-charts/
  ```

### 2. Update Helm chart repo

Update your Helm chart repositories to ensure that you are using the RisingWave Helm chart:

  ```bash
  helm repo update
  ```

>   If you are using AWS EKS, you also need to set it as the default cluster for Helm:

Reviewer 2: ```suggestion
  If you are using AWS EKS, you also need to update the local config for kubectl and Helm to access: 
```

  ```bash
  aws eks update-kubeconfig --name <your_eks_cluster_name>
  ```

### 3. Customize configuration

You can optionally customize your configuration for the RisingWave deployment by editing the [`values.yml`](https://github.com/risingwavelabs/helm-charts/blob/main/charts/risingwave/values.yaml) file.

> #### Customize meta backends

Reviewer: Please add a sentence about what the meta backend does. It stores the metadata of a RisingWave cluster.

   
For meta backends, we recommend using `etcd` by setting `tags.etcd=true`. See [`Configuration`](https://github.com/risingwavelabs/helm-charts/blob/main/docs/CONFIGURATION.md#customize-meta-store) for all the available meta backends.

> #### Customize state backends

Reviewer: Similarly, please provide a short decription of the state backend. The state backend in RisingWave serves as a fault-tolerante storage system for the state.

    - **Customize state store**: The state store in RisingWave serves as a fault-tolerant storage system for preserving system state. See [Configuration](https://github.com/risingwavelabs/helm-charts/blob/main/docs/CONFIGURATION.md#customize-state-store) for all the available options and [Examples](https://github.com/risingwavelabs/helm-charts/tree/main/examples/state-stores) for detailed usage of state stores.

    - **Bundled etcd and MinIO**: If you want to use `etcd` as the meta backend and `MinIO` as the state backend, the Helm chart for RisingWave offers the option to bundle them together. This allows for a quick and easy setup of the Helm chart. See [Configuration](https://github.com/risingwavelabs/helm-charts/blob/main/docs/CONFIGURATION.md#bundled-etcdminio-as-stores) for more details. To enable this feature, set `tags.bundle=true`.

5. Install the latest RisingWave Helm chart:

    ```bash

>     helm install --set wait=true <my-risingwave> risingwavelabs/risingwave

Reviewer 2: ```suggestion
    helm install -n risingwave --create-namespace --set wait=true -f values.yaml <my-risingwave> risingwavelabs/risingwave
```

    ```

    Where `<my-risingwave>` is the release name you choose to use for your RisingWave deployment. This command will install the latest stable version of RisingWave.

    If you want to install a particular version, you can specify the version via the `image-tag` attribute. Remember to replace `<version_number>` with the desired version, for example `v1.7.0`.

    ```bash

>     helm install --set wait=true --set image.tag=<version_number> <my-risingwave> risingwavelabs/risingwave

Reviewer 2: ```suggestion
    helm install -n risingwave --create-namespace --set wait=true --set image.tag=<version_number> <my-risingwave> -f values.yaml risingwavelabs/risingwave
```

    ```

    You may get an output message like this:

    ```bash
    NAME: my-risingwave
    LAST DEPLOYED: Wed Aug 16 15:35:19 2023
    NAMESPACE: default
    STATUS: deployed
    REVISION: 1
    TEST SUITE: None
    ```

6. Use the following command to check the deployment status:

  ```bash

>   kubectl get pods -l app.kubernetes.io/instance=<my-risingwave>

Reviewer 2: ```suggestion
  kubectl -n risingwave get pods -l app.kubernetes.io/instance=<my-risingwave>
```

  ```

  When your status looks like below, it means the RisingWave cluster starts successfully:

  ```bash
  NAME                                   READY   STATUS    RESTARTS        AGE
  risingwave-compactor-8dd799db6-hdjjz   1/1     Running   1 (8m33s ago)   11m
  risingwave-compute-0                   2/2     Running   0               11m
  risingwave-etcd-0                      1/1     Running   0               11m
  risingwave-frontend-7bd7b8c856-czdgd   1/1     Running   1 (8m33s ago)   11m
  risingwave-meta-0                      1/1     Running   0               11m
  risingwave-minio-5cfd8f5f64-6msqm      1/1     Running   0               11m
  ```

> ## Step 3: Access RisingWave

Reviewer 2: The command below needs to be 

```shell
kubectl -n risingwave port-forward svc/my-risingwave 4567:svc
```

to set the namespace.

## Optional: Customize your RisingWave deployment

> During installation or upgrade, you can customize your RisingWave deployment by providing the configuration file `values.yml`. You should edit the file before specifying it during installation or upgrade.

Reviewer 2: The paragraph is no longer necessary I think. Customization is mandatory now. Shall we move the content to step 4 above?

      memory: 64Mi
```

> ## Optional: Resize a node

Reviewer 2: There is also a [paragraph](https://github.com/risingwavelabs/helm-charts/blob/main/docs/CONFIGURATION.md#customize-pods-of-different-components) about the pod's configuration in the doc. Shall we add a link here?
